backend/src/Altinn.Authorizationadmin/Altinn.Authorizationadmin/Services/Implementation/PolicyAdministrationPoint.cs

```diff
@@ -230,17 +248,25 @@ private async Task<bool> WriteDelegationPolicyInternal(string policyPath, List<R
                         return false;
                     }
 
+                    ServiceResource resource = null;
+                    if (!string.IsNullOrWhiteSpace(resourceRegistryId))
+                    {
+                        resource = await _resourceRegistryClient.GetResource(resourceRegistryId);
+                    }
+                    
                     // Write delegation change to postgresql
                     DelegationChange change = new DelegationChange
                     {
                         DelegationChangeType = DelegationChangeType.Grant,
-                        AltinnAppId = $"{org}/{app}",
+                        AltinnAppId = altinnAppId,
                         OfferedByPartyId = offeredByPartyId,
                         CoveredByPartyId = coveredByPartyId,
                         CoveredByUserId = coveredByUserId,
                         PerformedByUserId = delegatedByUserId,
                         BlobStoragePolicyPath = policyPath,
-                        BlobStorageVersionId = blobResponse.Value.VersionId
+                        BlobStorageVersionId = blobResponse.Value.VersionId,
+                        ResourceId = resourceRegistryId != null ? resourceRegistryId : string.Empty,
+                        ResourceType = resource != null ? resource.ResourceType.ToString() : string.Empty
```

Comment:
For app delegations when resourceRegistryId and ResourceType is null then they should be set to null in the database also. not empty string.